toolchain/driver/driver.cpp

```diff
@@ -22,286 +25,481 @@
+)""",
+  };
 
-auto Driver::RunFullCommand(llvm::ArrayRef<llvm::StringRef> args) -> bool {
-  StreamDiagnosticConsumer stream_consumer(error_stream_);
-  DiagnosticConsumer* consumer = &stream_consumer;
-  std::unique_ptr<SortingDiagnosticConsumer> sorting_consumer;
-  // TODO: Figure out a command-line support library, this is temporary.
-  if (!args.empty() && args[0] == "-v") {
-    args = args.drop_front();
-    // Note this implies streamed output in order to interleave.
-    vlog_stream_ = &error_stream_;
-  } else if (!args.empty() && args[0] == "--print-errors=streamed") {
-    args = args.drop_front();
-  } else {
-    sorting_consumer = std::make_unique<SortingDiagnosticConsumer>(*consumer);
-    consumer = sorting_consumer.get();
+  enum class Phase {
+    Tokenize,
+    Parse,
+    Syntax,
+    Lower,
+    CodeGen,
+  };
+
+  friend auto operator<<(llvm::raw_ostream& out, Phase phase)
+      -> llvm::raw_ostream& {
+    switch (phase) {
+      case Phase::Tokenize:
+        out << "tokenize";
+        break;
+      case Phase::Parse:
+        out << "parse";
+        break;
+      case Phase::Syntax:
+        out << "syntax";
+        break;
+      case Phase::Lower:
+        out << "lower";
+        break;
+      case Phase::CodeGen:
+        out << "codegen";
+        break;
+    }
+    return out;
   }
 
-  if (args.empty()) {
-    error_stream_ << "ERROR: No subcommand specified.\n";
-    return false;
+  void Build(Args::CommandBuilder& b) {
+    b.AddStringPositionalArg(
+        {
+            .name = "FILE",
+            .help = R"""(
+The input Carbon source file to compile.
+)""",
+        },
+        [&](auto& arg_b) {
+          arg_b.Required(true);
+          arg_b.Set(&input_file_name);
+        });
+
+    b.AddOneOfOption(
+        {
+            .name = "phase",
+            .help = R"""(
+Selects the compilation phase to run. These phases are always run in sequence,
+so every phase before the one selected will also be run. The default is to
+compile, lower, and generate machine code.
+)""",
+        },
+        [&](auto& arg_b) {
+          arg_b.SetOneOf(
+              {
+                  arg_b.OneOfValue("tokenize", Phase::Tokenize),
+                  arg_b.OneOfValue("parse", Phase::Parse),
+                  arg_b.OneOfValue("syntax", Phase::Syntax),
+                  arg_b.OneOfValue("lower", Phase::Lower),
+                  arg_b.OneOfValue("codegen", Phase::CodeGen).Default(true),
+              },
+              &phase);
+        });
+
+    b.AddStringOption(
+        {
+            .name = "output",
+            .value_name = "FILE",
+            .help = R"""(
+The output filename for codegen.
+
+When this is a file name, either textual assembly or a binary object will be
+written to it based on the flag `--asm-output`. The default is to write a binary
+object file.
+
+This can also be the string `-` which will write the output to stdout. In that
```

Comment:
```suggestion
Passing `--output=-` will write the output to stdout. In that
```

Maybe a shorter/clearer way to say the same.